### PR TITLE
Partial Implementation of Torrent Client

### PR DESCRIPTION
**Pull request by badstreff**
Changes to the structure of the project:
bencoding package was moved to the TorrentClient package.
torrent.py was moved to the TorrentClient package.
sample_torrents/ was moved the the TorrentClient package.

The previous modules were left in place (for compatibility purposes) but future work on them should cease and begin on the new modules.

### CODE REVIEW

**Comment by virtualhealthcitizen** [comment]
I like it.

-- snek :snake:

**Comment by samliska** [comment]
Looks good to me

File: debug/TorrentClient/message.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        Message.__init__(self)
+        self.length = 13
+        self.id = 6
+        self.index = index
+        self.begin = begin
+        self.request_length = length
+
+    def to_bytes(self):
+        r = super(Request, self).to_bytes()
+        r += pack('!l', self.index)
```
Comment (virtualhealthcitizen) [inline debug/TorrentClient/message.py:274]: From https://docs.python.org/3/library/struct.html:

"_The form '!' is available for those poor souls who claim they can’t remember whether network byte order is big-endian or little-endian._"

Hah :P

Comment (badstreff) [reply debug/TorrentClient/message.py:274]: It's always big-endian, I just like typing '!' vs '>' :)

**Comment by badstreff** [comment]
This is completed enough to merge. Writing to disk will be a separate PR.

**Pull request merged**